src/components/CPU/BreakpointPanel.js

```diff
@@ -0,0 +1,167 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import '../ext/react-modal.css';
+import './BreakpointPanel.css';
+import {toString08X} from "../../utils/format";
+import {ContextMenu, ContextMenuTrigger, MenuItem} from "react-contextmenu";
+import listeners from "../../utils/listeners";
+
+export default function BreakpointPanel(props) {
+	const [cpuBreakpoints, setCpuBreakpoints] = useState([]);
+	const [memoryBreakpoints, setMemoryBreakpoints] = useState([]);
+
+	const fetchBreakpoints = useCallback(() => {
+		props.ppsspp.send({
+			event: 'cpu.breakpoint.list',
+		}).then(({breakpoints}) => {
+			setCpuBreakpoints(breakpoints);
+		});
+		props.ppsspp.send({
+			event: 'memory.breakpoint.list',
+		}).then(({breakpoints}) => {
+			setMemoryBreakpoints(breakpoints);
+		});
+	}, [props.ppsspp]);
+
+	useEffect(() => {
+		fetchBreakpoints();
+		// TODO cpu.breakpoint.* not sent by PPSSPP?
+		const listeners_ = listeners.listen({
+			'cpu.stepping': () => {
+				fetchBreakpoints();
+			},
+			// TODO memory.breakpoint add/update/remove
+			'cpu.breakpoint.add': () => {
+				fetchBreakpoints();
+			},
+			'cpu.breakpoint.update': () => {
+				fetchBreakpoints();
+			},
+			'cpu.breakpoint.remove': () => {
+				fetchBreakpoints();
+			},
+		});
+		return () => listeners.forget(listeners_);
+	}, [fetchBreakpoints]);
+
+	return (
+		<React.Fragment>
+			<ContextMenuTrigger id="addBreakpoint" holdToDisplay={-1}>
+				<table className="BreakpointPanel__table">
+					<thead>
+						{/* TODO Left align header, first column sizing, row selection */}
+						<tr>
+							<th>&nbsp;</th>

```

Comment:
nit: Should be fine without an nbsp?  But maybe this heading should be "Break".

-[Unknown]